application/plugin_base/item_list.py

```diff
@@ -1647,6 +1665,19 @@ def destroy_providers(self):
 		for path, provider in self._providers.items():
 			provider.release_archive_handle()
 
+	def create_intermediate_archive_provider(self, path=None):
```

Comment:
Additional note: the method create_intermediate_archive_provider() starts with full path (e.g. /home/foo/bar.zip/subdir1), checks if current path is supported by archive provider and creates the provider if it doesn't exist yet. If not it recursively strips of filenames from the end of the path and checks them as well (until it reaches / or an existing provider).